modules/apps/layout/layout-reports-web/src/main/resources/META-INF/resources/js/components/IssueDetail.js

```diff
@@ -60,3 +69,137 @@ HtmlPanel.propTypes = {
 	content: PropTypes.string.isRequired,
 	title: PropTypes.string.isRequired,
 };
+
+const FailingElementsPanel = ({failingElements, issueType}) => {
+	const [shownElements, setShownElements] = useState(10);
+
+	const normalizedFailingElements = normalizeFailingElements(
+		failingElements,
+		issueType
+	);
+
+	const totalElements = normalizedFailingElements.length;
+
+	const onViewMore = () => {
+		const newShownElements = shownElements + 10;
+
+		setShownElements(
+			newShownElements < totalElements ? newShownElements : totalElements
+		);
+	};
+
+	return (
+		<ClayPanel
+			collapsable
+			collapseClassNames="c-mb-4 c-mt-3"
+			defaultExpanded
+			displayTitle={
+				<span className="c-inner" tabIndex="-1">
+					<ClayPanel.Title>
+						<ClayLayout.ContentRow>
+							<ClayLayout.ContentCol
+								className="align-self-center panel-title"
+								expand
+							>
+								{Liferay.Language.get('failing-elements')}
+							</ClayLayout.ContentCol>
+							<ClayLayout.ContentCol>
+								<ClayBadge
+									displayType={
+										totalElements === 0 ? 'success' : 'info'
+									}
+									label={
+										totalElements >= 100
+											? '+100'
+											: totalElements
+									}
+								/>
+							</ClayLayout.ContentCol>
+						</ClayLayout.ContentRow>
+					</ClayPanel.Title>
+				</span>
+			}
+			displayType="unstyled"
+			showCollapseIcon={true}
+		>
+			<ClayPanel.Body>
+				<ClayList>
+					{normalizedFailingElements
+						.filter((element, index) => index < shownElements)
```

Comment:
It is a bit weird to use a filter to get the first elements of an array, why not using `slice` ?

Reply:
You are right 🙂 I'll change it!